src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -100,35 +96,13 @@ protected Process createProcess() throws IOException {
     }
 
     private Process createRunasProcess(String... commands) throws IOException {
+        // Expect username in format: DOMAIN\UserName
         String username = userDecorator.getUser();
-
-        byte[] credBlob = WindowsCredUtils.read(username);
-        ByteBuffer bb = ByteBuffer.wrap(credBlob);
-        CharBuffer cb = WindowsCredUtils.getCharsetForSystem().decode(bb);
-
-        // Prepend runas arguments to use it for running commands as different user
-        List<String> args = new ArrayList<>();
-        args.add(runasExePath);
-        args.add("-u:" + username);  // runas username
-        args.add("-p:" + cb);  // plain text password. TODO revisit this because it exposes plaintext password
-        args.add("-l:off");  // disable logging
-        args.add("-b:0"); // set exit code base number to 0
-        args.add("-w:" + dir); // set workdir explicitly
-        // runas is un-escaping customer-provided quotes and escape characters, so we need to escape them
-        // first so they unwrap correctly.
-        List<String> cmd = Arrays.stream(commands).map((s) ->
-                s.replace("\\\"", "\\\\\"").replace("\"", "\\\""))
-                .collect(Collectors.toList());
-        args.addAll(cmd);
-
-        Arrays.fill(cb.array(), (char) 0);  // zero-out temporary buffers
-        Arrays.fill(bb.array(), (byte) 0);
-
-        ProcessBuilder pb = new ProcessBuilder();
-        pb.environment().putAll(environment);
-        Process p = pb.directory(dir).command(args).start();
-        args.clear();  // best effort to clear password presence
-        return p;
+        ProcessBuilderForWin32 winPb = new ProcessBuilderForWin32();
+        winPb.environment().clear();
+        winPb.environment().putAll(environment);
+        winPb.user(username, new String(getPassword(username)));
```

Comment:
comment for future readers: we discussed pushing something that works and will re-visit using a char[] instead of the string password so we can clear it out of memory by zeroing out the array as soon as possible without relying on garbage collection